linalg/slepc.hpp

```diff
@@ -69,39 +75,92 @@ class SlepcEigenSolver
+      @param[in] i The index for the eigenvalue you want ordered by SlepcEigenSolver::SetWhichEigenpairs
+      @param[out] lr The real component of the eigenvalue
+      @param[out] lc The imaginary component of the eigenvalue
+      @note the index @a i must be between 0 and SlepcEigenSolver::GetNumConverged - 1
+   */
    void GetEigenvalue(unsigned int i, double & lr, double & lc) const;
 
-   /// Get the corresponding eigenvector
+   /** @brief Get the ith eigenvector after the system has been solved
+      @param[in] i The index for the eigenvector you want ordered by SlepcEigenSolver::SetWhichEigenpairs
+      @param[out] vr The real components of the eigenvector
+      @note the index @a i must be between 0 and SlepcEigenSolver::GetNumConverged - 1
+   */
    void GetEigenvector(unsigned int i, Vector & vr) const;
+
+   /** @brief Get the ith eigenvector after the system has been solved
+      @param[in] i The index for the eigenvector you want ordered by SlepcEigenSolver::SetWhichEigenpairs
+      @param[out] vr The real components of the eigenvector
+      @param[out] vc The imaginary components of the eigenvector
+      @note the index @a i must be between 0 and SlepcEigenSolver::GetNumConverged - 1
+   */
    void GetEigenvector(unsigned int i, Vector & vr, Vector & vc) const;
 
-   /// Target spectrum for the eigensolver. Target imaginary is not supported
-   /// without complex support in SLEPc, and intervals are not implemented.
+   /** @brief Target spectrum for the eigensolver.
+
+       This will define the order in which the eigenvalues/eigenvectors are indexed
+       after the call to SlepcEigenSolver::Solve.
+       @note Target imaginary is not supported without complex support in SLEPc,
+       and intervals are not implemented.
+   */
    enum Which
    {
+      /// The eigenvalues with the largest complex magnitude (default)
       LARGEST_MAGNITUDE,
+      /// The eigenvalues with the smallest complex magnitude
       SMALLEST_MAGNITUDE,
+      /// The eigenvalues with the largest real component
       LARGEST_REAL,
+      /// The eigenvalues with the smallest real component
       SMALLEST_REAL,
+      /// The eigenvalues with the largest imaginary component
       LARGEST_IMAGINARY,
+      /// The eigenvalues with the smallest imaginary component
       SMALLEST_IMAGINARY,
+      /// The eigenvalues with complex magnitude closest to the target value
       TARGET_MAGNITUDE,
+      /// The eigenvalues with the real component closest to the target value
       TARGET_REAL
    };
 
+   /** @brief Spectral transformations that can be used by the solver in order
+      to accelerate the convergence to the target eignevalues
+   */
    enum SpectralTransformation
    {
+      /// Utilize the shift of origin strategy
       SHIFT,
+      /// Utilize the shift and invert strategy
       SHIFT_INVERT
    };
 
+   /** @brief Set the which eignevalues the solver will target and the order they will be indexed in
```

Comment:
minor:"
- "Set the which" -> "Set which" ?
- spelling: "eignevalues" -> "eigenvalues"